src/test/db_benchmark/common.ml

```diff
@@ -0,0 +1,117 @@
+open Core
+
+(* Configuration constants - configurable via environment variables *)
+let keys_per_block =
+  Sys.getenv "KEYS_PER_BLOCK" |> Option.value_map ~default:125 ~f:int_of_string
+
+let value_size =
+  Sys.getenv "VALUE_SIZE"
+  |> Option.value_map ~default:(128 * 1024) ~f:int_of_string
+(* 128 KB *)
+
+let warmup_blocks =
+  Sys.getenv "WARMUP_BLOCKS" |> Option.value_map ~default:800 ~f:int_of_string
+
+(* Fixed seed for reproducibility *)
+let random_seed = 42
+
+(* Get key ID from block number and offset within block *)
+let key_of_block_offset block_num offset = (block_num * keys_per_block) + offset
+
+(* Get all key IDs for a given block *)
+let keys_of_block block_num =
+  List.init keys_per_block ~f:(fun offset ->
+      key_of_block_offset block_num offset )
+
+(* Generate random data for a value *)
+let generate_value () =
+  let random_state = Random.State.make [| random_seed |] in
+  String.init value_size ~f:(fun _ ->
+      Char.of_int_exn (Random.State.int random_state 256) )
+
+(* Cache a single value to avoid regenerating it every time *)
+let cached_value = lazy (generate_value ())
+
+(* Get the cached value *)
+let get_value () = Lazy.force cached_value
```

Comment:
We're using a same value for all blocks. I wonder if these backend would have some optimizations that make the performance better, it's better to use distinct values for distinct keys.